File: app/src/main/java/com/example/services/authenticator/AuthUtils.java
package com.example.services.authenticator;

import java.util.Calendar;
import java.util.Date;

final class AuthUtils {
  static boolean isAccessTokenExpired(long expiresIn) {
    Calendar c = Calendar.getInstance();
    Date nowDate = c.getTime();

    c.setTimeInMillis(expiresIn);
    Date expireDate = c.getTime();

    int result = nowDate.compareTo(expireDate);

    return result == -1;
  }
}

File: app/src/main/kotlin/com/example/services/authenticator/AuthUtils.kt
package com.example.services.authenticator

import java.util.*

internal object AuthUtils {
  fun isAccessTokenExpired(expiresIn: Long): Boolean {
    val c = Calendar.getInstance()
    val nowDate = c.time

    c.timeInMillis = expiresIn
    val expireDate = c.time

    val result = nowDate.compareTo(expireDate)

    return result == -1
  }
}
